### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by adamwalach**
This PR is a 
 - [ ] Feature Implementation
 - [ ] Bug Fix
 - [x] Documentation

**Which issue(s) this PR fixes** 
Fixes #https://github.com/kyma-project/kyma/issues/4481

### CODE REVIEW

**Review by jasiu001 — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
 
 Afterward, see the topics below.
 
+**Note:** Between versions 0.2.0 and 0.3.0 Service Catalog changed its internal storage mechanism.
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
**Note:** Between versions 0.2.0 and 0.3.0, Service Catalog changed its internal storage mechanism.
```

File: docs/README.md [hunk 2]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
...
+Versions 0.2.0 and older used its own API-Server and etcd.
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
Versions 0.2.0 and older used its own API Server and etcd. 
```

File: docs/README.md [hunk 3]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
 
 Afterward, see the topics below.
 
+**Note:** Between versions 0.2.0 and 0.3.0 Service Catalog changed its internal storage mechanism.
+Versions 0.2.0 and older used its own API-Server and etcd. 
+
+Starting from version 0.3.0 Service Catalogs moved to Custom Resource Definitions (native K8S feature).
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
Starting from version 0.3.0, Service Catalogs moved to Custom Resource Definitions solution, which is a native K8S feature.
```

File: docs/README.md [hunk 4]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
...
+
+The API-Server implementation will be supported (only by providing bug fixes) for next 9 months.
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
The API Server implementation will be supported by providing bug fixes for the next 9 months.
```

File: docs/README.md [hunk 5]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
...
+If you are an existing user of version 0.2.0 please see the [migration guide](./migration-apiserver-to-crds.md)
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
If you still use Service Catalog version 0.2.0, read the [migration guide](./migration-apiserver-to-crds.md)
```

File: docs/design.md [hunk 6]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
 ![Service Catalog Design](images/sc-design.svg)
 
 The above is the high level architecture of Service Catalog.
+Service Catalog has two basic building blocks: a webhook server and a controller.
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
Service Catalog has two basic building blocks: a Webhook Server and a controller.
```

File: docs/design.md [hunk 7]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
 
-### API Server
+### Webhook Server
 
-The API Server is an HTTPS
-[REST](https://en.wikipedia.org/wiki/Representational_state_transfer)ful 
-front-end for [etcd](https://coreos.com/etcd/) 
-(we can implement more storage backends, but we haven't done so)
+The Webhook Server makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
+to manage custom resources. Admission Webhooks is a feature available in the Kubernetes API server,
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
to manage custom resources. Admission Webhook is a feature available in the Kubernetes API Server, 
```

File: docs/design.md [hunk 8]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
-### API Server
+### Webhook Server
 
-The API Server is an HTTPS
-[REST](https://en.wikipedia.org/wiki/Representational_state_transfer)ful 
-front-end for [etcd](https://coreos.com/etcd/) 
-(we can implement more storage backends, but we haven't done so)
+The Webhook Server makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
+to manage custom resources. Admission Webhooks is a feature available in the Kubernetes API server, 
+that allows to implement arbitrary control decisions like validation or mutation.
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
that allows you to implement arbitrary control decisions, such as validation or mutation.
```

File: docs/design.md [hunk 9]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
-design).
 
-### API Server
+### Webhook Server
 
-The API Server is an HTTPS
-[REST](https://en.wikipedia.org/wiki/Representational_state_transfer)ful 
-front-end for [etcd](https://coreos.com/etcd/) 
-(we can implement more storage backends, but we haven't done so)
+The Webhook Server makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
The Webhook Server uses [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
```

File: docs/design.md [hunk 10]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
-(via the
-[Kubernetes API aggregator](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/)
-to perform [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations on 
-Service Catalog resources (the ones listed in the previous section). For example, when a user runs 
-`kubectl get clusterservicebroker`, they will be talking to the Service
-Catalog API server to get the list of `ClusterServiceBroker` resources.
-
-The current version of all Service Catalog API resources is `v1beta1`.
-You can see the structure of each resource in detail at
+For every resource managed by Service Catalog (the ones listed in the previous section) there is a separate handler defined. You can see the structure at
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
For every resource managed by Service Catalog (the ones listed in the previous section), there is a separate handler defined. You can see the structure at 
```

File: docs/design.md [hunk 11]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
-to perform [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations on 
-Service Catalog resources (the ones listed in the previous section). For example, when a user runs 
-`kubectl get clusterservicebroker`, they will be talking to the Service
-Catalog API server to get the list of `ClusterServiceBroker` resources.
-
-The current version of all Service Catalog API resources is `v1beta1`.
-You can see the structure of each resource in detail at
+For every resource managed by Service Catalog (the ones listed in the previous section) there is a separate handler defined. You can see the structure at 
+[`pkg/webhook/servicecatalog`](https://github.com/kyma-incubator/service-catalog/tree/crds/pkg/webhook/servicecatalog).
+The current version of all Service Catalog API resources is `v1beta1`, they are all defined here:
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
The current version of all Service Catalog API resources is `v1beta1`. The resources are defined here:
```

File: docs/design.md [hunk 12]
```diff
@@ -97,40 +97,30 @@ the basic concepts of the OSB API.
...
 [`pkg/apis/servicecatalog/v1beta1/types.go`](https://github.com/kubernetes-sigs/service-catalog/blob/master/pkg/apis/servicecatalog/v1beta1/types.go).
 
+If you want to learn more about Admission Webhooks read [this](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) guide.
```
Comment (klaudiagrz) [inline docs/design.md]:
```suggestion
If you want to learn more about Admission Webhooks, read [this](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) document.
```

File: docs/design.md (hunk 10 above)
Comment (klaudiagrz) [reply docs/design.md]: I do not understand the part in ( ).

File: docs/devguide.md [hunk 13]
```diff
@@ -124,10 +124,10 @@ layout:
     │   └── catalog             # Helm chart for deploying the service catalog
     │   └── ups-broker          # Helm chart for deploying the user-provided service broker
     ├── cmd                     # Contains "main" Go packages for each service catalog component binary
-    │   └── apiserver           # The service catalog API server service-catalog command
     │   └── controller-manager  # The service catalog controller manager service-catalog command
     │   └── service-catalog     # The service catalog binary, which is used to run commands
     │   └── svcat               # The command-line interface for interacting with kubernetes service-catalog resources
+    │   └── webhook             # The service catalog webhook server webhook command
```
Comment (klaudiagrz) [inline docs/devguide.md]: Is it correct?

File: docs/devguide.md [hunk 14]
```diff
@@ -334,7 +334,8 @@ functionality or introduce instability.  See [FeatureGates](feature-gates.md)
 for more details.
 
 When adding a FeatureGate to Helm charts, define the variable
-`fooEnabled` with a value of `false` in [values.yaml](https://github.com/kubernetes-sigs/service-catalog/blob/master/charts/catalog/values.yaml).  In the [API Server](https://github.com/kubernetes-sigs/service-catalog/blob/master/charts/catalog/templates/apiserver-deployment.yaml) and [Controller](https://github.com/kubernetes-sigs/service-catalog/blob/master/charts/catalog/templates/controller-manager-deployment.yaml)
+`fooEnabled` with a value of `false` in [values.yaml](https://github.com/kubernetes-sigs/service-catalog/blob/master/charts/catalog/values.yaml).
```
Comment (klaudiagrz) [inline docs/devguide.md]:
```suggestion
`fooEnabled` with the `false` value in [values.yaml](https://github.com/kubernetes-sigs/service-catalog/blob/master/charts/catalog/values.yaml).  
```

File: docs/install.md [hunk 15]
```diff
@@ -3,15 +3,10 @@ title: Install
...
-[API Aggregator](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/),
-which is a specialized proxy server that sits in front of the core API Server.
-
-Service Catalog provides an API server that sits behind the API aggregator,
-so you'll be using `kubectl` as normal to interact with Service Catalog.
-
-To learn more about API aggregation, please see the
-[Kubernetes documentation](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/).
+Service Catalog requires Kubernetes at version 1.11 or higher. 
+Starting from version 0.2.x Service Catalog makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)
```
Comment (klaudiagrz) [inline docs/install.md]:
```suggestion
Starting from version 0.2.x, Service Catalog uses [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
```

File: docs/install.md [hunk 16]
```diff
@@ -3,15 +3,10 @@ title: Install
...
-
-Service Catalog provides an API server that sits behind the API aggregator,
-so you'll be using `kubectl` as normal to interact with Service Catalog.
-
-To learn more about API aggregation, please see the
-[Kubernetes documentation](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/).
+Service Catalog requires Kubernetes at version 1.11 or higher. 
+Starting from version 0.2.x Service Catalog makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
+to manage custom resources. It uses [Additional Printer Columns](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/#additional-printer-columns) 
+so you can use `kubectl` as normal to interact with Service Catalog.
```
Comment (klaudiagrz) [inline docs/install.md]:
```suggestion
so you can use `kubectl` to interact with Service Catalog.
```

File: docs/devguide.md (hunk 13 above)
Comment (adamwalach) [reply docs/devguide.md]: I changed it to `The service catalog webhook server command`

File: docs/README.md [hunk 17]
```diff
@@ -14,6 +14,14 @@ This page is an index for the articles in here. We recommend you start by readin
 
 Afterward, see the topics below.
 
+**Note:** Between versions 0.2.0 and 0.3.0, Service Catalog changed its internal storage mechanism.
+Versions 0.2.0 and older used its own API Server and etcd. 
+
+Starting from version 0.3.0, Service Catalogs moved to Custom Resource Definitions solution, which is a native K8S feature.
+
+The API Server implementation will be supported by providing bug fixes for the next 9 months.
+If you still use Service Catalog version 0.2.0, read the [migration guide](./migration-apiserver-to-crds.md)
```
Comment (klaudiagrz) [inline docs/README.md]:
```suggestion
If you still use Service Catalog version 0.2.0, read the [migration guide](./migration-apiserver-to-crds.md).
```

File: docs/install.md [hunk 18]
```diff
@@ -3,15 +3,10 @@ title: Install
...
-Kubernetes 1.9 or higher clusters run the
-[API Aggregator](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/),
-which is a specialized proxy server that sits in front of the core API Server.
-
-Service Catalog provides an API server that sits behind the API aggregator,
-so you'll be using `kubectl` as normal to interact with Service Catalog.
-
-To learn more about API aggregation, please see the
-[Kubernetes documentation](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/).
+Service Catalog requires Kubernetes at version 1.11 or higher.
```
Comment (klaudiagrz) [inline docs/install.md]:
```suggestion
Service Catalog requires Kubernetes version 1.11 or higher. 
```

**Review by klaudiagrz — APPROVED** [review]

File: docs/README.md [hunk 19]
```diff
@@ -27,7 +35,6 @@ Afterward, see the topics below.
 - [Code conventions](./code-standards.md)
 - [Developer guide](./devguide.md)
 - [Design](./design.md)
-- [Notes on authentication](./auth.md)
```
Comment (mszostok) [inline docs/README.md]: should be reverted

Comment (adamwalach) [reply docs/README.md]: reverted

**Pull request merged**